### PR TITLE
fix multiple findings regarding serverside-rejection/exceptions

### LINKED ISSUE #213: Caching-store exception propagation
if a e.g. RequestMemory.add/.put fails on the server side, the returned promise is not rejected.
furthermore Cache.put contains invalid references in the exception-case.

`this.getIdentity`

is not valid in this context.
environment: latest dgrid,dstore, dojo 1.13

### PR DESCRIPTION
**Pull request by gerpres**
propagate exception to promise. use correct reference for local store.
fixes #213

### CODE REVIEW

File: Cache.js [hunk 1]
```diff
@@ -152,9 +152,12 @@ define([
 				// the result from the put should be dictated by the master store and be unaffected by the cachingStore,
 				// unless the master store doesn't implement put
 				return result || cachedPutResult;
-			}, function () {
+			}, function (exception) {
 				// Assuming a rejection of a promise invalidates the local cache
-				cachingStore.remove((directives && directives.id) || this.getIdentity(object));
+				cachingStore.remove((directives && directives.id) || cachingStore.getIdentity(object));
+				if(exception) {
```
Comment (edhager) [inline Cache.js]: This is propagating the rejection only when the promise was rejected with a value or object.  Wouldn't we want it to always throw an error?

File: Cache.js [hunk 2]
```diff
@@ -152,9 +152,12 @@ define([
...
+					throw exception;
```
Comment (edhager) [inline Cache.js]: Would it be possible for you to update the `Cache` unit test so it covers this?

File: Cache.js (hunk 1 above)
Comment (gerpres) [reply Cache.js]: agree. I added an additional 'throw'

File: Cache.js (hunk 2 above)
Comment (gerpres) [reply Cache.js]: I have no prior experience with the-intern - but I did my best updating the cache-testcases

**Comment by gerpres** [comment]
fixed another finding in _fetch() and added some basic test-cases as requested. pls review again

**Comment by dylans** [comment]
@edhager please take another look when you have a moment, thanks!

**Comment by edhager** [comment]
Sorry it took me so long to get back to this...

I make some cosmetic changes: converted double quotes to single quotes and modified the code formatting a bit.

Landed in commit eaff671d252986b30c58a03e7aeb60c577dc3ae1.

Thanks @gerpres!

**Pull request closed without merging**